script/job_generator/generate_job_yaml.py

```diff
@@ -70,44 +74,142 @@ def job_prefix_coding(job: Dict[str, Tuple[str, str]]) -> str:
 
 
 @typechecked
-def job_image(job: Dict[str, Tuple[str, str]], container_version: float) -> str:
-    """Generate the image url deppending on the host and device compiler and the selected backend
-    of the job.
+def get_images_from_registry(container_version: str) -> List[str]:
+    """Returns list of all container images for a given tag.
+
+    Args:
+        container_version (str): Container tag
+
+    Returns:
+        List[str]: list of container images
+    """
+    # hard coding the url and project ID is fine, because there is only one
+    # container registry
+    gl = gitlab.Gitlab(url="https://codebase.helmholtz.cloud")
+    project = gl.projects.get("4742")
+    repositories = project.repositories.list(get_all=True)
+
+    # uniform data structure
+    if not isinstance(repositories, list):
+        repositories = [repositories]
+
+    images = []
+    # this process is actual pretty slow
+    # maybe it does a HTML request each time if we iterate over repo.tags.list()
+    # and get a new tag
+    for repo in repositories:
+        for tag in repo.tags.list():
+            if tag.attributes["name"] == container_version:
+                images.append(tag.attributes["location"])
+
+    return images
+
+
+@typechecked
+def job_image(
+    job: Dict[str, Tuple[str, str]],
+    container_version: float,
+    gitlab_images: List[str] = {},
+) -> str:
+    """Generate the image url depending on the host and device compiler and the
+    selected backend of the job.
+
+    If gitlab_images is not empty, only images that are exist are returned.
+    Otherwise, the name of the image is constructed by a set of rules. In this
+    case, the image may not exist. If it looks for an existing image and it does
+    not exist, it returns an image that best matches the software's requirements.
```

Comment:
```suggestion
    If gitlab_images is not empty only existing images are returned.
    Otherwise, the name of the image is constructed by a set of rules. However,
    the resulting image may not exist. In this case an image is returned that best
    matches the software's requirements.
```